xla/service/gpu/fusions/triton/triton_fusion_emitter_legacy_matmul.cc

```diff
@@ -637,10 +637,13 @@ absl::StatusOr<Value> EmitBroadcast(ImplicitLocOpBuilder& b,
   Value expanded_input = tensor_input;
   int dim_idx = 0;
   for (const DimProperties& dim : side.tiled_dims) {
-    if (auto* spec = analysis->IterSpec(side.scope, &broadcast, dim.index);
-        spec != nullptr && spec->at(0).stride > 0) {
-      if (analysis->IterSpec(side.scope, broadcast.operand(0), dim.index) ==
-          nullptr) {
+    const auto* output_spec =
+        analysis->IterSpec(side.scope, &broadcast, dim.index);
+    if (output_spec != nullptr && output_spec->at(0).stride > 0) {
+      const auto* input_spec =
+          analysis->IterSpec(side.scope, broadcast.operand(0), dim.index);
+      if (input_spec == nullptr ||
+          output_spec->at(0).count > input_spec->at(0).count) {
```

Comment:
Could you clarify when that is true, both in a comment and in the CL message?

It's not quite clear what "more broadcasts" means here. Can you qualify the "more"?

Reply:
Done.